app/src/main/java/com/github/onlynotesswent/ui/common/Popups.kt

```diff
@@ -137,6 +149,108 @@ fun CreationDialog(
             }
       })
 }
+/**
+ * Displays a popup dialog to browse and navigate the file system.
+ *
+ * This popup allows users to view folders in a hierarchical structure. Users can navigate into
+ * subfolders, view folder contents, and return to the root folder.
+ *
+ * @param onDismiss Callback invoked when the popup is dismissed.
+ * @param folderViewModel The ViewModel that provides the folder data.
+ */
+@Composable
+fun FileSystemPopup(onDismiss: () -> Unit, folderViewModel: FolderViewModel) {
+  var selectedFolder by remember { mutableStateOf<Folder?>(folderViewModel.selectedFolder.value) }
+  var folderSubFolders by remember { mutableStateOf<List<Folder>>(emptyList()) }
+  val userRootFolders = folderViewModel.userRootFolders.collectAsState()
+
+  Dialog(onDismissRequest = { onDismiss() }) {
+    Box(
+        modifier =
+            Modifier.testTag("FileSystemPopup")
+                .fillMaxWidth(0.9f) // Adjust the popup width
+                .fillMaxHeight(0.5f)
+                .padding(16.dp)
+                .background(
+                    color = MaterialTheme.colorScheme.background,
+                    shape = RoundedCornerShape(12.dp))) {
+          Column(modifier = Modifier.fillMaxWidth().fillMaxHeight()) {
+            Box(
+                modifier =
+                    Modifier.fillMaxWidth()
+                        .background(MaterialTheme.colorScheme.primary)
+                        .padding(vertical = 12.dp, horizontal = 16.dp)) {
+                  Text(
+                      text =
+                          if (selectedFolder == null) "Folders in: root"
+                          else "Folders in: ${selectedFolder!!.name}",
+                      style = MaterialTheme.typography.bodyMedium,
+                      color = MaterialTheme.colorScheme.onPrimary)
+                }
+
+            Column(
+                modifier =
+                    Modifier.weight(1f)
+                        .padding(start = 16.dp, end = 16.dp, top = 8.dp)
+                        .verticalScroll(rememberScrollState()),
+                verticalArrangement = Arrangement.spacedBy(4.dp)) {
+                  if (selectedFolder == null) {
```

Comment:
To improve readability, creating a Boolean Mutable State (called root for instance) containing whether selected folder is null could be added.